daemon/api.go

```diff
@@ -186,6 +187,12 @@ var (
 		POST:   postBuy,
 	}
 
+	readyToBuyCmd = &Command{
+		Path:   "/v2/buy/ready",
```

Comment:
The reasoning behind this is outlined in the following document. But basically boils down to the UX being "check the user's account is set up (TOS/credit card)", followed by a "buy X from Y for $$?" then actually trying to make the purchase. Of course the actual "/v2/buy" API may also return the same "tos-not-accepted/no-payment-method" errors, to handle a client that forgoes the "pre flight check".

https://docs.google.com/document/d/1Gv4m2zB2zaR_MbBRGKI_JJ6cXKVHeEs94PfSiVbLGF0/edit#heading=h.xk6xwgv1x0k6